007-extension-flag.md

```diff
@@ -0,0 +1,57 @@
+| Title  | Implement --extension flag  |
+|--------|-----------------------------|
+| Author | @bmeck                      |
+| Status | DRAFT                       |
+| Date   | 2017-08-09T10:00:00-05:00   |
+
+## Description
+
+The Node CLI currently accepts input in both filename (`node app.js`) and opaque text stream form (`node -e '123'` and `node <app.js`). This behavior always results in running the stream as a CommonJS goal. With the upcoming usage of ESM and WASM there will be cases where developers wish to pipe other forms of text to the CLI.
+
+This proposal is to implement an `--entry-extension` CLI flag for the cases which may want to use a stream that does not have a file extension.
```

Comment:
If we are defining the name here, might be worthwhile to also specify `--ext` as a shorter-hand. People will want to see using it as reasonable UX.

Reply:
do we have any abbreviates `--` flags?

Reply:
Unsure but I don't really see an issue with it.

Reply:
What about `-x` ?

Reply:
Could use that, I suppose. Does it have any common UNIX meaning it might be confused with?

Reply:
no , http://tldp.org/LDP/abs/html/standard-options.html